WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginEmailFragment.java

```diff
@@ -0,0 +1,397 @@
+
+        setHasOptionsMenu(true);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        ViewGroup rootView = (ViewGroup) inflater.inflate(R.layout.login_email_screen, container, false);
+
+        mEmailEditText = (EditText) rootView.findViewById(R.id.login_email);
+        mEmailEditText.addTextChangedListener(this);
+
+        mEmailEditTextLayout = (TextInputLayout) rootView.findViewById(R.id.login_email_layout);
+
+        mNextButton = (Button) rootView.findViewById(R.id.login_email_next_button);
+        mNextButton.setOnClickListener(mNextClickListener);
+        mUsernamePasswordButton = rootView.findViewById(R.id.login_email_username_password);
+
+        mEmailEditText.setOnFocusChangeListener(new View.OnFocusChangeListener() {
+            public void onFocusChange(View v, boolean hasFocus) {
+                if (!hasFocus) {
+                    autocorrectEmail();
+                }
+            }
+        });
+
+        autofillFromBuildConfig();
+
+        mEmailEditText.setOnEditorActionListener(new TextView.OnEditorActionListener() {
+            @Override
+            public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
+                if ((didPressNextKey(actionId, event) || didPressEnterKey(actionId, event))) {
+                    next(getEmail());
+                    return true;
+                } else {
+                    return false;
+                }
+            }
+        });
+
+        mUsernamePasswordButton.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (mLoginNavInputEmail != null) {
+                    mLoginNavInputEmail.loginViaUsernamePassword();
+                }
+            }
+        });
+
+        return rootView;
+    }
+
+    @Override
+    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        Toolbar toolbar = (Toolbar) view.findViewById(R.id.toolbar);
+        ((AppCompatActivity) getActivity()).setSupportActionBar(toolbar);
+
+        ActionBar actionBar = ((AppCompatActivity) getActivity()).getSupportActionBar();
+        if (actionBar != null) {
+            actionBar.setDisplayShowTitleEnabled(false);
+            actionBar.setDisplayHomeAsUpEnabled(true);
+        }
+
+        if (savedInstanceState != null) {
+            boolean isInProgress = savedInstanceState.getBoolean(KEY_IN_PROGRESS);
+            if (isInProgress) {
+                showEmailCheckProgressDialog();
+            }
+        }
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        if (context instanceof LoginStateGetter.FsmGetter) {
+            mLoginNavInputEmail = ((LoginStateGetter.FsmGetter) context).getLoginStateGetter().getLoginNavInputEmail();
```

Comment:
`getLoginStateGetter()` cannot be null unless we've made some error with the LoginNavFragment initialisation (see also https://github.com/wordpress-mobile/WordPress-Android/pull/5848#discussion_r116673071). It is OK to crash in that case since that would be unrecoverable anyway and we should notice it ASAP via Fabric.

Then, `getLoginNavInputEmail()` can also throw if the navigation state is not in the email state. That's by design as well. It's also an unrecoverable situation since it would reveal a mismatch between the screen and the navigation state handler's state.